From 9604dba2dadb77fe4dce2c536c44ceba56c3a05c Mon Sep 17 00:00:00 2001
From: NIIBE Yutaka <gniibe@fsij.org>
Date: Mon, 7 Nov 2016 11:49:08 +0900
Subject: [PATCH 38/52] Fix path of scdaemon.

* am/cmacros.am (AM_CPPFLAGS): Add GNUPG_LIBEXECDIR.
* configure.ac (GNUPG_DEFAULT_SCD): Use GNUPG_LIBEXECDIR.

--
Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>
---
 am/cmacros.am | 3 ++-
 configure.ac  | 2 +-
 src/scd/scd.c | 3 +--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/am/cmacros.am b/am/cmacros.am
index ca24616..7cfab3f 100644
--- a/am/cmacros.am
+++ b/am/cmacros.am
@@ -1,4 +1,5 @@
-AM_CPPFLAGS += -DGNUPG_BINDIR="\"$(bindir)\""
+AM_CPPFLAGS += -DGNUPG_BINDIR="\"$(bindir)\"" \
+               -DGNUPG_LIBEXECDIR="\"$(libexecdir)\""
 datadir = @datadir@
 localedir = $(datadir)/locale
 DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
diff --git a/configure.ac b/configure.ac
index 5e87907..1886e2c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -69,7 +69,7 @@ AC_DEFINE_UNQUOTED(NEED_KSBA_VERSION, "$NEED_KSBA_VERSION",
 AH_BOTTOM([
 /* Setup the hardwired names of modules. */
 #ifndef GNUPG_DEFAULT_SCD
-#define GNUPG_DEFAULT_SCD    ( GNUPG_BINDIR "/scdaemon" )
+#define GNUPG_DEFAULT_SCD    ( GNUPG_LIBEXECDIR "/scdaemon" )
 #endif
 #ifndef GNUPG_DEFAULT_DIRMNGR
 #define GNUPG_DEFAULT_DIRMNGR    ( GNUPG_BINDIR "/dirmngr" )
diff --git a/src/scd/scd.c b/src/scd/scd.c
index 9f60026..abd63f9 100644
--- a/src/scd/scd.c
+++ b/src/scd/scd.c
@@ -180,8 +180,7 @@ scd_connect (scd_context_t *scd_ctx, const char *scd_path,
       no_close_list[i] = -1;
 
       /* connect to the scdaemon and perform initial handshaking */
-      rc = assuan_pipe_connect (&assuan_ctx, scd_path, argv,
-                                no_close_list);
+      rc = assuan_pipe_connect (&assuan_ctx, scd_path, argv, no_close_list);
       if (!rc)
 	{
 	  log_msg_debug (loghandle,
-- 
2.17.0

